jujugui/static/gui/src/app/components/help/tour/test-tour.js

```diff
@@ -0,0 +1,52 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const React = require('react');
+
+const Tour = require('./tour');
+const SvgIcon = require('../../svg-icon/svg-icon');
+
+const jsTestUtils = require('../../../utils/component-test-utils');
+
+describe('Tour', () => {
+  it('renders', () => {
+    const renderer = jsTestUtils.shallowRender(
+      <Tour close={sinon.stub()} endTour={sinon.stub()} />,
+      true
+    );
+
+    const output = renderer.getRenderOutput();
+
+    expect(output.props.children[0]).toEqualJSX(
+      <span className="back-to-help"
+        onClick={sinon.stub()}>
+        <SvgIcon
+          className="back-to-help__icon"
+          name="chevron_down_16"
+          size="16"
+        />
+        Back to GUI help
+      </span>
+    );
+
+    assert.deepEqual(output.props.children[1].props.extraClasses, ['tour']);
+
+    // Slides
+    const slides = output.props.children[1].props.children;
+    assert.isDefined(slides);
+    assert.isDefined(slides[0]);
+    assert.deepEqual(slides[0].props.className, 'tour__slide');
+    assert.isDefined(slides[1]);
+    assert.deepEqual(slides[1].props.className, 'tour__slide');
```

Comment:
Well ok I guess...But this will still allow more slides to be added and pass, but removed slides will fail...seems like a fragile test...maybe it would be less work to add a length check and put these in a loop (although you've already copypasta) :)

Reply:
Loops are for chumps, copy and paste is king.
Will improve.